iceoryx_posh/test/integrationtests/service_discovery/test_roudi_findservice.cpp

```diff
@@ -32,54 +32,154 @@ class RoudiFindService_test : public RouDiServiceDiscoveryTest
+    this->InterOpWait();
+
+    ASSERT_EQ(false, isServiceOffered);
+}
+
+TEST_F(RoudiFindService_test, OfferServiceWithAnyEventIdFails)
+{
+    auto isServiceOffered =
+        senderRuntime->offerService(iox::capro::ServiceDescription(123u, 456u, iox::capro::AnyEvent));
+    this->InterOpWait();
+
+    ASSERT_EQ(false, isServiceOffered);
+}
+
+TEST_F(RoudiFindService_test, OfferServiceWithAnyEventIdStringFails)
+{
+    auto isServiceOffered = senderRuntime->offerService(
+        iox::capro::ServiceDescription({"service1", "instance1", iox::capro::AnyEventString}));
+    this->InterOpWait();
+
+    ASSERT_EQ(false, isServiceOffered);
+}
+
+TEST_F(RoudiFindService_test, ReofferedServiceWithValidServiceDescriptionCanBeFound)
+{
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+    senderRuntime->stopOfferService({"service1", "instance1"});
+    this->InterOpWait();
     senderRuntime->offerService({"service1", "instance1"});
     this->InterOpWait();
-    iox::runtime::InstanceContainer instanceContainer;
-    receiverRuntime->findService({"service1", "instance1"}, instanceContainer);
 
-    ASSERT_THAT(instanceContainer.size(), Eq(1u));
-    ASSERT_THAT(*instanceContainer.begin(), Eq(IdString_t("instance1")));
+    auto instanceContainer = receiverRuntime->findService({"service1", "instance1"});
+
+    ASSERT_THAT(instanceContainer.value().size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.value().begin(), Eq(IdString_t("instance1")));
+}
+
+TEST_F(RoudiFindService_test, OfferExsistingServiceMultipleTimesIsRedundant)
+{
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+
+    auto instanceContainer = receiverRuntime->findService({"service1", "instance1"});
+
+    ASSERT_THAT(instanceContainer.value().size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.value().begin(), Eq(IdString_t("instance1")));
+}
+
+TEST_F(RoudiFindService_test, FindSameServiceMultipleTimesReturnsSingleInstance)
+{
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+
+    auto instanceContainer = receiverRuntime->findService({"service1", "instance1"});
+    ASSERT_THAT(instanceContainer.value().size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.value().begin(), Eq(IdString_t("instance1")));
+
+    instanceContainer = receiverRuntime->findService({"service1", "instance1"});
+    ASSERT_THAT(instanceContainer.value().size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.value().begin(), Eq(IdString_t("instance1")));
 }
 
 TEST_F(RoudiFindService_test, DISABLED_OfferMultiMethodServiceSingleInstance_PERFORMANCETEST42)
 {
     senderRuntime->offerService({"service1", "instance1"});
     senderRuntime->offerService({"service2", "instance1"});
     senderRuntime->offerService({"service3", "instance1"});
```

Comment:
Could copy this test and make this `"instance1_1"`, `"instance2_1"` and `"instance3_1"` or `"instance_a"`, `"instance_b"`, `"instance_c"`? Just to catch the error that findService returns the instances of the wrong serviceId

Reply:
current implementation for mentioned scenario doesn't return error, it returns empty container

Reply:
This is not regarding returning an error, but catching an error. As far as I can tell, there is no check for `findService({"service2", "instance1"})` looking in `service1` for that instance. Since the instances have all the same name, this is an untested error case. If the instances have different names in one test, it could be checked that the returned container is not empty.

Reply:
@elBoberido sorry but i dint get the test you are asking for, what i got is 

    senderRuntime->offerService({"service1", "instance1_1"});
    senderRuntime->offerService({"service2", "instance2_1"});
    senderRuntime->offerService({"service3", "instance3_1"});

and then what combination you want in findService()  ??
is it 
findService({"service1", "instance1_1"});
findService({"service2", "instance2_1"});
findService({"service3", "instance3_1"});

and then check for each returned container for non empty?

Reply:
let's assume we have the following services offered
```
senderRuntime->offerService({"service1", "instance1"});
senderRuntime->offerService({"service2", "instance2"});
```
a weird bug in `findService` could ignore the passed service ID and `findService({"service2", "instance1"});`would return `instance1`. When all instances have the same name, this error case is not found. So this would be a `OfferMultiMethodServiceWithDistinctSingleInstance`

Reply:
i think that is not the case, following test passes

    senderRuntime->offerService({"service1", "instance1"});
    senderRuntime->offerService({"service2", "instance2"});
    this->InterOpWait();

    auto instanceContainer = receiverRuntime->findService({"service1", "instance1"});
    ASSERT_THAT(instanceContainer.value().size(), Eq(1u));
    ASSERT_THAT(*instanceContainer.value().begin(), Eq(IdString_t("instance1")));

    instanceContainer = receiverRuntime->findService({"service2", "instance1"});
    ASSERT_THAT(instanceContainer.value().size(), Eq(0u));

    instanceContainer = receiverRuntime->findService({"service2", "instance2"});
    ASSERT_THAT(instanceContainer.value().size(), Eq(1u));
    ASSERT_THAT(*instanceContainer.value().begin(), Eq(IdString_t("instance2")));

Reply:
I would assume that test to pass, but where can I find it? It seems it's not in this file.

Reply:
@elBoberido i pushed it now